frontend/tests/playwright/fixtures/metrics-reporter.ts

```diff
@@ -0,0 +1,135 @@
+import {
+   FullResult,
+   Reporter,
+   TestCase,
+   TestResult,
+} from '@playwright/test/reporter';
+import { execSync } from 'child_process';
+import { v4 as uuidv4 } from 'uuid';
+import fs from 'fs';
+
+interface ResultParams {
+   id: string;
+   test_name: string;
+   time_start: number;
+   time_end: number;
+   time_taken: number;
+   pass: string;
+   test_suite_id: string | null;
+   file_path: string | null;
+}
+
+class MetricsReporter implements Reporter {
+   branch: string;
+   commit: string;
+   test_suite_id: string;
+   results: Array<Object>;
+   timeStart!: number;
+   timeEnd!: number;
+
+   constructor() {
+      this.branch =
+         process.env.BUILD_BRANCH ||
+         execSync('git rev-parse --abbrev-ref HEAD', {
+            encoding: 'utf-8',
+         }).trim();
+      this.commit =
+         process.env.BUILD_COMMIT ||
+         execSync('git rev-parse HEAD', { encoding: 'utf-8' }).trim();
+      this.test_suite_id = uuidv4();
+
+      this.results = [];
+   }
+
+   onBegin(): void {
+      this.timeStart = Date.now();
+   }
+
+   getTestName(test: TestCase) {
+      return test.titlePath().slice(3).join(' -> ');
+   }
+
+   getStatus(status: string): number {
+      if (status === 'passed') {
+         return 1;
+      } else if (status === 'interrupted') {
+         return -1;
+      } else {
+         return 0;
+      }
+   }
+
+   addResult({
+      id,
+      test_name,
+      time_start,
+      time_end,
+      time_taken,
+      pass,
+      test_suite_id,
+      file_path,
+   }: ResultParams) {
+      this.results.push({
+         id: id,
+         test_name: test_name,
+         time_start: time_start / 1000,
+         time_end: time_end / 1000,
+         time_taken: time_taken / 1000,
+         pass: this.getStatus(pass),
+         test_suite_id: test_suite_id,
+         file_path: file_path,
+         commit: this.commit,
+         branch: this.branch,
+      });
+   }
+
+   onTestEnd(test: TestCase, result: TestResult) {
+      if (result.status !== 'skipped') {
+         this.addResult({
+            id: uuidv4(),
+            test_name: this.getTestName(test),
+            time_start: result.startTime.valueOf(),
+            time_end: result.startTime.valueOf() + result.duration,
+            time_taken: result.duration,
+            pass: result.status,
+            test_suite_id: this.test_suite_id,
+            file_path: test.location.file,
+         });
+      }
+   }
+
+   // Add the total time for the test suite took to run
+   addFullResult(fullResult: FullResult) {
+      this.addResult({
+         id: this.test_suite_id,
+         test_name: 'ci-playwright',
+         time_start: this.timeStart,
+         time_end: this.timeEnd,
+         time_taken: this.timeEnd - this.timeStart,
```

Comment:
Btw, this is super weird, but it's how we do it. I _think_ this is where we should use the QUEUED_AT. And I think we only do this for the full result.

Ref: https://github.com/iFixit/ifixit/blob/master/Tests/Exec/continuous-integration-wrapper.sh#L126-L130

Reply:
```
mysql> select * from test_runs where test_name = 'ci' limit 1 \G;
*************************** 1. row ***************************
           id: b6d3181e-e0d9-4996-aff9-a4e1eb121e04
    test_name: ci
   time_start: 1661735424
     time_end: 1661735679
   time_taken: 1661735371
         pass: 1
test_suite_id: NULL
    file_path: https://github.com/iFixit/ifixit/runs/8061985604?check_suite_focus=true
       commit: 95226e06a18596d7679cfc4ae015e4d29670cf8c
       branch: ci-metrics--delete-old-test-metrics
```

Reply:
For individual test runs, we set the `time_taken` to actually be the time take:

```
mysql> select * from test_runs limit 1 \G;
*************************** 1. row ***************************
           id: 000004de-3a52-4aea-85d4-93bc63b79e1a
    test_name: UtmTrackerServiceTest -> testUtmTrackerTimestampConversion
   time_start: 1686674319.992
     time_end: 1686674320.908
   time_taken: 0.916
         pass: 1
test_suite_id: 005f2b7a-5a93-4682-875f-9e46fe5d5282
    file_path: Tests/Integration/Cart/UtmTrackerServiceTest.php
       commit: 6cff69f9518ab677ec99708f9059251302cd923f
       branch: add-rubocop-to-ci
```

Reply:
Oh... I was thinking that the time taken for the whole test-suite would be the start of playwright test and the end.

However, if we use the `QUEUED_AT` for the `time_start`, would that mean it would include the time of the steps between the `setup-ci-env-var` action and `Run Playwright Tests` step?

Reply:
> Btw, this is super weird, but it's how we do it.  

I agree. Set the `time_taken` for the whole test-suite as `QUEUED_AT` in https://github.com/iFixit/react-commerce/pull/1773/commits/170799c0ddb8fb77f36cc1866acbd6589625472e.